### ISSUE TITLE
Post provide a Generic Type instead of object

### ISSUE THREAD

**Issue by mglodack** [awseward/restivus#14]
I did have a slight enhancement to the post method and possibly others that exposed a delegate to creating `HttpContent`.

I had to cast my object for [this](https://github.com/awseward/restivus/blob/master/src/Restivus/RestClient.cs#L37) Func<object, HttpContent>. Any way we can make that a generic. Something like `Func<T, HttpContent>`?

**Comment by awseward** [comment]
Yeah I don't think there's any reason we couldn't do that. Probably just oversight on my part to not go that way originally.

Good call
